test/unit/org/apache/cassandra/index/sai/cql/intersection/RandomIntersectionTester.java

```diff
@@ -72,14 +74,15 @@ public void createTableAndIndexes()
         createIndex("CREATE INDEX ON %s(s1) USING 'sai'");
         createIndex("CREATE INDEX ON %s(s2) USING 'sai'");
 
-        numPartitions = nextInt(15000, 100000);
+        numRows = nextInt(16000, 24000);
```

Comment:
This is the most important part of the change. Everything else is just making sure (via some histograms and observation) that we still have both queries that match nothing and queries that match varying numbers of rows for all test scenarios/subclasses.